roles/common/tasks/check_upgrades.yml

```diff
@@ -20,54 +20,56 @@
         installed_components.acs is defined and
         installed_components.amps is defined and
         installed_components.api_explorer is defined and
-        ( 
+        (
           acs.edition != installed_components.acs.edition or
           acs.version.split('.')[:3] | join('.') != installed_components.acs.version.split('.')[:3] | join('.') or
           acs.version is version(installed_components.acs.version, 'lt') or
-          amps != installed_components.amps or
-          api_explorer != installed_components.api_explorer
+          amps.aos_module.version != installed_components.amps.aos_module.version or
+          amps.device_sync.version != installed_components.amps.device_sync.version or
+          amps.googledrive_repo.version != installed_components.amps.googledrive_repo.version or
+          amps.googledrive_share.version != installed_components.amps.googledrive_share.version or
```

Comment:
I am only a little unsure about checking the single amps that are not really mandatory: it will fail if user remove one amp and won't check any additional amp that we don't provide by default.

so ideally it should be something like a foreach on amps checking that version is the same or was not installed before